stock_app.py

```diff
@@ -80,19 +90,19 @@ def analyze_stock(self):
         try:
             self.stock_data = self.fetch_stock_data(stock_symbol, start_date, end_date)
             self.analyze_and_plot(stock_symbol, self.stock_data)
-        except Exception as e:
-            messagebox.showerror("Error", f"Error fetching data: {str(e)}")
+        except yf.errors.YFinanceError as e:  # Catch specific exception
+            messagebox.showerror("Error", f"Error fetching data: {e!s}")
             return
 
     def fetch_stock_data(self, symbol, start_date, end_date):
         return yf.download(symbol, start=start_date, end=end_date)
 
     def analyze_and_plot(self, symbol, data):
-        df = data.reset_index()[['Date', 'Close']]
-        df = df.rename(columns={'Date': 'ds', 'Close': 'y'})
+        df = data.reset_index()[["Date", "Close"]]
+        df = df.rename(columns={"Date": "ds", "Close": "y"})
```

Comment:
```suggestion
        date_close = data.reset_index()[["Date", "Close"]]
        date_close = date_close.rename(columns={"Date": "ds", "Close": "y"})
```
Repeat below.  `date_close_df` or `date_close_dataframe` would also be acceptable.